### ISSUE TITLE
Aggregate all the pre releases

### ISSUE THREAD

**Issue by radoslavirha** [release-it/conventional-changelog#62]
Hi,

is there any way how to aggregate pre releases in changelog and create single (minor version) entry?

Using `release-it minor --git.tagExclude='*[-]*'` generates changelog with each pre release.

Ideal flow:
```
release-it minor // e.g. 1.1.0
release-it minor --preRelease=beta // 1.2.0-beta.0 => commit, tag, gitlab release but skip changelog?
release-it --preRelease // 1.2.0-beta.1 => commit, tag, gitlab release but skip changelog?
release-it minor // 1.2.0 => commit,tag,release and create aggregated changelog from 1.1.0 to 1.2.0 (do not show pre releases)
```

or update changelog on pre releases, and ideally remove those entries on minor release and create aggregated changelog?

Thanks

**Comment by webpro** [comment]
Unfortunately not. This sounds very custom, you'll have to manually update the changelog. I also think the efforts to make this happen reliably isn't really worth it, so I'm going to close this issue. Feel free to follow up with a PR, though, and I'm happy to look into it.

**Comment by ChAyLom** [comment]
I don't think this is relevant yet, but I have faced a similar issue. I found a solution that may help others who encounter the same issue.

#### TLDR: I fixed this by setting `writerOpts.generateOn` option to `null`.

This problem occurs due to the default behavior of [conventional-changelog-writer](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-writer). To be exact, the option [`generateOn`](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-writer#generateon) by default set to [func](https://github.com/conventional-changelog/conventional-changelog/blob/6adf8a0b5213988d06800df847f4636184c90186/packages/conventional-changelog-writer/src/options.ts#L67), which return `true` if a commit has a valid semver `version` property, which [come from](https://github.com/conventional-changelog/conventional-changelog/blob/6adf8a0b5213988d06800df847f4636184c90186/packages/conventional-changelog-core/lib/merge-config.js#L89) tag (tag is always provided, regardless `git.tagExclude`). And if `generateOn` returns `true`, `conventional-changelog-writer` strip current version in changelog, and continues with just founded version. Set this option to `null` is equivalent to `() => false`, which disable any grouping by version, and use only provided version for all commits.
Maybe I missed something, but in sum, I think my description corresponds to the [writer sources](https://github.com/conventional-changelog/conventional-changelog/blob/6adf8a0b5213988d06800df847f4636184c90186/packages/conventional-changelog-writer/src/writers.ts).

But I am dont commit my changelog, and use this plugin only for generate release notes of latest version. So if you are using this plugin to generate changelog for the all versions in your repo, you need to provide custom `generateOn` function. Which return `true` if commit has stable version, but `false` for prerelease.
